src/app/clusters/fan-control-server/fan-control-server.cpp

```diff
@@ -360,80 +372,70 @@ void MatterFanControlClusterServerAttributeChangedCallback(const app::ConcreteAt
     }
     case PercentSetting::Id: {
         DataModel::Nullable<Percent> percentSetting;
+        uint8_t speedMax;
         Status status = PercentSetting::Get(attributePath.mEndpointId, percentSetting);
         VerifyOrReturn(Status::Success == status && !percentSetting.IsNull());
 
+        status = SpeedMax::Get(attributePath.mEndpointId, &speedMax);
+        VerifyOrReturn(Status::Success == status,
+                       ChipLogError(Zcl, "Failed to get SpeedMax with error: 0x%02x", to_underlying(status)));
+
+        // Avoid circular callback calls
+        gPercentWriteInProgress = true;
         // If PercentSetting is set to 0, the server SHALL set the FanMode attribute value to Off.
         if (percentSetting.Value() == 0)
         {
             status = SetFanModeToOff(attributePath.mEndpointId);
-            VerifyOrReturn(Status::Success == status,
-                           ChipLogError(Zcl, "Failed to set FanMode to off with error: 0x%02x", to_underlying(status)));
+            VerifyOrDo(status == Status::Success,
```

Comment:
Same suggestion as below to use VerifyOrReturn with gSpeedWriteInProgress = false;

Reply:
Fixed too, thanks!